src/ign.cc

```diff
@@ -123,9 +125,27 @@ extern "C" IGNITION_GAZEBO_VISIBLE int runServer(const char *_sdfString,
     const char *_renderEngineServer, const char *_renderEngineGui,
     const char *_file, const char *_recordTopics)
 {
-  ignition::gazebo::ServerConfig serverConfig;
+  ignition::transport::Node node;
+  std::string starting_world_path{""};
+  std::condition_variable condition;
+  std::mutex mutex;  
+
+  // Create a subscriber just so we can check when the message has propagated
+  std::function<void(const ignition::msgs::StringMsg_V &)> topicCb =
+      [&starting_world_path, &mutex, &condition](const auto &_msg)
+      {
+        std::unique_lock<std::mutex> lock(mutex);
+        starting_world_path = _msg.data(0);
+        starting_world_path = starting_world_path.erase(0,7);
+        condition.notify_all();
+      };
+
+  std::unique_lock<std::mutex> lock(mutex);
+  node.Subscribe("/gazebo/starting_world", topicCb);
+  condition.wait(lock);
```

Comment:
I was able to avoid the freeze by unsubscribing from the topic after we got it:

```diff
diff --git a/src/ign.cc b/src/ign.cc
index 6985c7f96..863234c47 100644
--- a/src/ign.cc
+++ b/src/ign.cc
@@ -142,7 +142,12 @@ extern "C" IGNITION_GAZEBO_VISIBLE int runServer(const char *_sdfString,
 
   std::unique_lock<std::mutex> lock(mutex);
   node.Subscribe("/gazebo/starting_world", topicCb);
+
+  igndbg << "Waiting for a world to be set from the GUI..." << std::endl;
   condition.wait(lock);
+  node.Unsubscribe("/gazebo/starting_world");
+  ignmsg << "Received world [" << starting_world_path << "] from the GUI."
+         << std::endl;
 
   // Path for logs
   ignition::gazebo::ServerConfig serverConfig;
```

I suspect we can get more than one message and end up in a locked state.